src/node_platform.cc

```diff
@@ -0,0 +1,73 @@
+#include "node_platform.h"
+
+namespace node {
+
+NodePlatform::NodePlatform(int thread_pool_size, uv_loop_t* loop,
+                           v8::TracingController* tracing_controller)
+    : threads_(thread_pool_size), loop_(loop) {
+  if (tracing_controller) {
+    tracing_controller_.reset(tracing_controller);
+  } else {
+    v8::TracingController* controller = new v8::TracingController();
+    tracing_controller_.reset(controller);
+  }
+}
+
+NodePlatform::~NodePlatform() {}
+
+size_t NodePlatform::NumberOfAvailableBackgroundThreads() {
+  return static_cast<size_t>(threads_);
+}
+
+void RunBackgroundTask(uv_work_t *req) {
```

Comment:
Maybe these internal functions could be made `static` or part of an anonymous namespace?